mailpoet/lib/WooCommerce/MultichannelMarketing/MPMarketingChannel.php

```diff
@@ -108,4 +120,15 @@ public function get_supported_campaign_types(): array { // phpcs:ignore PSR1.Met
   public function get_campaigns(): array { // phpcs:ignore PSR1.Methods.CamelCapsMethodName.NotCamelCaps
     return []; // will be updated in MAILPOET-5698
   }
+
+  /**
+   * Whether the task is completed.
+   * If the setting 'version' is not null it means the welcome wizard
+   * was already completed so we mark this task as completed as well.
+   */
+  protected function isMPSetupComplete(): bool {
```

Comment:
I wonder if we should reuse (or create if one doesn't exist already) a more generic method to check if the MP setup is completed? My concern here is that we probably need to check this in other places and things might break if, in the future, we change the logic of the setup and use a criteria other than whether or not `version` exists to determine if the setup is completed or not. What do you think?

Reply:
Thank you for the suggestion, and I agree with you.
For the two places ([1](https://github.com/mailpoet/mailpoet/blob/1a0d3ece134e2a4079a9842147befff253524bb7/mailpoet/lib/WooCommerce/MailPoetTask.php), [2](https://github.com/mailpoet/mailpoet/blob/1a0d3ece134e2a4079a9842147befff253524bb7/mailpoet/lib/WooCommerce/MultichannelMarketing/MPMarketingChannel.php)) we need to run these checks, the parent classes do not pass through the MailPoet DI.
How would you suggest we go about implementing it?

Reply:
Good point. In this case, maybe the best approach is to use `ContainerWrapper::getInstance()->get()`? What do you think?

Reply:
The scope of this might be a bit larger than I expected. I've created another ticket to handle the implementation: https://mailpoet.atlassian.net/browse/MAILPOET-5736.

Reply:
Sounds good to me! Thanks!

Is there anything else you would like to do before a final review? Asking as the PR is still assigned to you.

I will be AFK tomorrow, so I will leave the PR approved in case you want to move it to QA tomorrow. Have to further check anything when I'm back.

Reply:
> Is there anything else you would like to do before a final review? Asking as the PR is still assigned to you.

Nope.

> I will be AFK tomorrow, so I will leave the PR approved in case you want to move it to QA tomorrow. Have to further check anything when I'm back.

Thank you, @rodrigoprimo 

I'll move the PR to QA.